src/librustc_typeck/check/closure.rs

```diff
@@ -179,6 +179,9 @@ fn deduce_expectations_from_obligations<'a,'tcx>(
                 ty::Predicate::TypeOutlives(..) => None,
                 ty::Predicate::WellFormed(..) => None,
                 ty::Predicate::ObjectSafe(..) => None,
+                ty::Predicate::ClosureKind(_closure_def_id, kind) => {
+                    return Some(kind);
```

Comment:
@nikomatsakis a note: changing this to expressing `None` (as opposed to returning) causes the stage 2 `librustc_typeck` compile to go through without adversely affecting tests. Not sure if that's 'right' though.